cmd/agent.go

```diff
@@ -81,22 +97,20 @@ func (ar *AgentRunner) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	if daemon == true {
-		ar.runDaemon(plugins, policyBundles)
-	} else {
-		err := ar.runInstance(plugins, policyBundles)
-
+	if !daemon {
+		err := ar.runInstance(plugins, policyBundles, nc)
 		if err != nil {
 			ar.logger.Error("error running instance", "error", err)
 			return err
 		}
 	}
+	ar.runDaemon(plugins, policyBundles, nc)
```

Comment:
New bug here. If daemon == false, it will runInstance(), and if no error is returned, it will start the daemon, because there is no return :)